From 9add0e0309af7cae02319466795a1b41b37ab6cf Mon Sep 17 00:00:00 2001
From: stakiar <stakiar@cloudera.com>
Date: Wed, 31 May 2017 16:18:15 -0700
Subject: [PATCH 1160/1164] CLOUDERA-BUILD: CDH-53684: HIVE-16665: Fixing bug
 in Utilities.java for String vs Path handling

Change-Id: I8d7a6c0a939e2b749d0f856699eaa222983f49cc
(cherry picked from commit 6c86b0ee93065bcabbcab24a89215a412b06a79a)
---
 .../org/apache/hadoop/hive/ql/exec/Utilities.java  |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/exec/Utilities.java b/ql/src/java/org/apache/hadoop/hive/ql/exec/Utilities.java
index 8abcb0c..4496f5f 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/exec/Utilities.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/exec/Utilities.java
@@ -3542,7 +3542,7 @@ private GetInputPathsCallable(Path path, JobConf job, MapWork work, Path hiveScr
     @Override
     public Path call() throws Exception {
       if (!this.skipDummy && isEmptyPath(this.job, this.path, this.ctx)) {
-        return createDummyFileForEmptyPartition(this.path, this.job, this.work.getPathToPartitionInfo().get(this.path),
+        return createDummyFileForEmptyPartition(this.path, this.job, this.work.getPathToPartitionInfo().get(this.path.toString()),
                 this.hiveScratchDir);
       }
       return this.path;
@@ -3610,11 +3610,11 @@ private static Path createDummyFileForEmptyPartition(Path path, JobConf job, Par
   private static void updatePathForMapWork(Path newPath, MapWork work, Path path) {
     // update the work
     if (!newPath.equals(path)) {
-      PartitionDesc partDesc = work.getPathToPartitionInfo().get(path);
-      work.getPathToAliases().put(newPath.toString(), work.getPathToAliases().get(path));
-      work.getPathToAliases().remove(path);
+      PartitionDesc partDesc = work.getPathToPartitionInfo().get(path.toString());
+      work.getPathToAliases().put(newPath.toString(), work.getPathToAliases().get(path.toString()));
+      work.getPathToAliases().remove(path.toString());
 
-      work.getPathToPartitionInfo().remove(path);
+      work.getPathToPartitionInfo().remove(path.toString());
       work.getPathToPartitionInfo().put(newPath.toString(), partDesc);
     }
   }
-- 
1.7.9.5

